### DISCUSSION TITLE
使用llvm::json::Array时遇到的问题

### DISCUSSION THREAD

**Discussion by yinhan-rq** [arcsysu/SYsU-lang#21] [Q&A] [answered]
在实验中，我想用自己定义的一个`llvm::json::Array`去替换`stak.push_back(llvm::json::Object{ {"inner", llvm::json::Array{inner}}})`中的`llvm::json::Array{inner}`。
但是得到了这样的报错：
`error: no matching constructor for initialization of 'llvm::json::Object'
            stak.push_back(llvm::json::Object{{"inner", newArray},`
`newArray`的定义如下：
`llvm::json::Array newArray`。
 看报错好像是Object中没有对应的构造函数，但是我不太清楚为什么`llvm::json::Array`就不会产生这样的错误。

**Comment by wu-kan** [comment, marked as answer]
这我也不清楚…既然原来的写法能work，不如就保持原来的写法？
